### PR TITLE
lc102, lc637, lc107

### PR DESCRIPTION
**Pull request by hetobias**
did lc102, lc637, lc107, updated README

### CODE REVIEW

File: tobiasHe/assignments/treesBFS/lc102/lc102.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		this.left = null;
+		this.right = null;
+	}
+}
+/* 
+  using BFS to traverse the tree, while keeping the values of each level
+  initialize a queue with root
+  keep track of queueLength, to move on to next levels
+  shift out the queue and check if current.value has any left and right children
+  if there are any, add them to the queue
```
Comment (MelissaPastore) [inline tobiasHe/assignments/treesBFS/lc102/lc102.js:17]: Nice overview but missing a step about adding the current level's values to the current Level array.

File: tobiasHe/assignments/treesBFS/lc102/lc102.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+};
+
+var root = new TreeNode(12);
+root.left = new TreeNode(7);
+root.right = new TreeNode(1);
+root.left.left = new TreeNode(9);
+root.right.left = new TreeNode(10);
+root.right.right = new TreeNode(5);
+console.log(`Level order traversal: ${traverse(root)}`);
+
```
Comment (MelissaPastore) [inline tobiasHe/assignments/treesBFS/lc102/lc102.js:51]: You could add some test cases here. Especially for edge cases.

File: tobiasHe/assignments/treesBFS/lc102/lc102.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}
+/* 
+  using BFS to traverse the tree, while keeping the values of each level
+  initialize a queue with root
+  keep track of queueLength, to move on to next levels
+  shift out the queue and check if current.value has any left and right children
+  if there are any, add them to the queue 
+  then push the value of that level into levels
+  return levels 
+
```
Comment (MelissaPastore) [inline tobiasHe/assignments/treesBFS/lc102/lc102.js:20]: Don't forget to outline any edge cases.

File: tobiasHe/assignments/treesBFS/lc107/lc107.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+}
+/* 
+  using BFS to traverse the tree, while keeping the values of each level
+  initialize a queue with root
+  keep track of queueLength, to move on to next levels
+  shift out the queue and check if currentNode.value has any left and right children
+  if there are any, add them to the queue 
+  then unshift the value of that level into levels  (because we are doing traversal in reverse order)
+  return levels 
+
```
Comment (MelissaPastore) [inline tobiasHe/assignments/treesBFS/lc107/lc107.js:20]: Don't forget to include edge cases and how you will handle them in your overview.

File: tobiasHe/assignments/treesBFS/lc107/lc107.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+};
+
+var root = new TreeNode(12);
+root.left = new TreeNode(7);
+root.right = new TreeNode(1);
+root.left.left = new TreeNode(9);
+root.right.left = new TreeNode(10);
+root.right.right = new TreeNode(5);
+console.log(`Reverse level order traversal: ${traverse(root)}`);
+
```
Comment (MelissaPastore) [inline tobiasHe/assignments/treesBFS/lc107/lc107.js:49]: You can also write more test cases, especially for any edge cases.

File: tobiasHe/assignments/treesBFS/lc637/lc637.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+}
+/* 
+  using BFS to traverse the tree, while keeping a sum value of each level
+  initialize a queue with root
+  keep track of queueLength, to move on to next levels
+  shift out the queue and check if current.value has any left and right children
+  if there are any, add them to the queue 
+  then push the average of that level into result
+  return result 
+
```
Comment (MelissaPastore) [inline tobiasHe/assignments/treesBFS/lc637/lc637.js:20]: Nice overview - but missing some steps like keeping track of the sum of the nodes in the level. Could also include some details about edge cases here.

File: tobiasHe/assignments/treesBFS/lc637/lc637.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+var root = new TreeNode(12);
+root.left = new TreeNode(7);
+root.right = new TreeNode(1);
+root.left.left = new TreeNode(9);
+root.left.right = new TreeNode(2);
+root.right.left = new TreeNode(10);
+root.right.right = new TreeNode(5);
+
+console.log(`Level averages are: ${find_level_averages(root)}`);
```
Comment (MelissaPastore) [inline tobiasHe/assignments/treesBFS/lc637/lc637.js:49]: Don't forget to add some additional test cases.

**Review by MelissaPastore — APPROVED** [review]
